src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -71,6 +71,41 @@
 }
 
 
+class IndexParent:
+    def __init__(self) -> None:
```

Comment:
Can we always init with values not None? if index._parent is None, then no parent, if it is not, it should mean parent exists.

Reply:
I think the _init_ can always take a df as input, index._parent can remain optional, which makes the relationship more clear

Reply:
done! The init should now take `parent` and be the only way to set the parent details.